storage/aptosdb/src/pruner/state_store/mod.rs

```diff
@@ -129,42 +122,31 @@ impl StateStorePruner {
     ///
     /// We issue (range) deletes on the index only periodically instead of after every pruning batch
     /// to avoid sending too many deletions to the DB, which takes disk space and slows it down.
-    fn maybe_purge_index(&self) -> anyhow::Result<()> {
+    fn maybe_purge_index(&self) -> Result<()> {
         const MIN_INTERVAL: Duration = Duration::from_secs(10);
         const MIN_VERSIONS: u64 = 60000;
 
+        let min_readable_version = self.min_readable_version.load(Ordering::Relaxed);
+        let index_purged_until_version = self.index_purged_until_version.load(Ordering::Relaxed);
+
         // A deletion is issued at most once in one minute and when the pruner has progressed by at
         // least 60000 versions (assuming the pruner deletes as slow as 1000 versions per second,
         // this imposes at most one minute of work in vain after restarting.)
         let now = Instant::now();
-        if self.min_readable_version.load(Ordering::Relaxed) < self.index_min_nonpurged_version() {
-            warn!("State pruner inconsistent, min_readable_version is {} and  index_min_non-purged_version is {}",
-                self.min_readable_version.load(Ordering::Relaxed), self.index_min_nonpurged_version());
-            return Ok(());
-        }
-
         if now - *self.index_purged_at.lock() > MIN_INTERVAL
-            && self.min_readable_version.load(Ordering::Relaxed)
-                - self.index_min_nonpurged_version()
-                + 1
-                > MIN_VERSIONS
+            && min_readable_version - index_purged_until_version > MIN_VERSIONS
         {
-            let new_min_non_purged_version = self.min_readable_version.load(Ordering::Relaxed) + 1;
             self.db.range_delete::<StaleNodeIndexSchema, Version>(
-                &self.index_min_nonpurged_version(),
-                &new_min_non_purged_version, // end is exclusive
+                &index_purged_until_version,
+                &(min_readable_version + 1), // end is exclusive
```

Comment:
Since the end is exclusive here and we want `min_readable_version` not to be deleted, this should be `&(min_readable_version` instead right?

Reply:
Ah I see. May be we can add a comment here for clarity.